plugins/catalog-backend/src/service/util.ts

```diff
@@ -166,3 +171,80 @@ export function expandLegacyCompoundRelationsInEntity(entity: Entity): Entity {
   }
   return entity;
 }
+
+/**
+ * Given a list of catalog processors, filter out the ones that are disabled
+ * through the `catalog.processorOptions` config and sort them by priority.
+ */
+export function filterAndSortProcessors(
+  processors: CatalogProcessor[],
+  config: Config,
+): CatalogProcessor[] {
+  function getProcessorOptions(
+    processor: CatalogProcessor,
+  ): Config | undefined {
+    const root = config.getOptionalConfig('catalog.processorOptions');
+    try {
+      return root?.getOptionalConfig(processor.getProcessorName());
+    } catch {
+      // We silence errors specifically here, to cover for cases where the
+      // processor name contains special characters which makes the config
+      // reader throw an error.
+      return undefined;
+    }
+  }
+
+  function getProcessorDisabled(processor: CatalogProcessor): boolean {
+    return (
+      getProcessorOptions(processor)?.getOptionalBoolean('disabled') === true
+    );
+  }
+
+  function getProcessorPriority(processor: CatalogProcessor): number {
+    let priority =
+      getProcessorOptions(processor)?.getOptionalNumber('priority');
+
+    if (priority === undefined) {
+      try {
+        priority = processor.getPriority?.();
+      } catch {
+        // In case the processor method throws, just return default priority
+      }
+    }
+
+    return priority ?? 20;
+  }
+
+  return processors
+    .filter(p => !getProcessorDisabled(p))
+    .sort((a, b) => getProcessorPriority(a) - getProcessorPriority(b));
+}
+
+/**
+ * Given a list of entity providers, filter out the ones that are disabled
+ * through the `catalog.providerOptions` config.
+ */
+export function filterProviders(
+  providers: EntityProvider[],
+  config: Config,
+): EntityProvider[] {
+  function getProviderOptions(provider: EntityProvider): Config | undefined {
+    const root = config.getOptionalConfig('catalog.providerOptions');
+    try {
+      return root?.getOptionalConfig(provider.getProviderName());
```

Comment:
Can we already use 
```ts
const [_, config] ?= root?.getOptionalConfig(provider.getProviderName());
return config;
```
?

Reply:
I ... don't like how that reads :D Yes you can use that operator though.

Reply:
I on the other land love this and the `??=` :D But I think it requires some newer ECMAScript support to use these